pkg/reconciler/instances/istio/istioctl/commander.go

```diff
@@ -3,21 +3,66 @@ package istioctl
 import (
 	"os"
 	"os/exec"
+
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/file"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	istioctlBinaryPathEnvKey = "ISTIOCTL_PATH"
 )
 
-//go:generate mockery -name=Commander
+//go:generate mockery --name=Commander --outpkg=mock --case=underscore
 // Commander for istioctl binary.
 type Commander interface {
 
 	// Install wraps istioctl installation command.
-	Install(istioCtlPath, istioOperatorPath, kubeconfigPath string) error
+	Install(istioOperator, kubeconfig string, logger *zap.SugaredLogger) error
+
+	// Upgrade wraps istioctl upgrade command.
+	Upgrade(istioOperator, kubeconfig string, logger *zap.SugaredLogger) error
+
+	// Version wraps istioctl version command.
+	Version(kubeconfig string, logger *zap.SugaredLogger) ([]byte, error)
 }
 
+var execCommand = exec.Command
+
 // DefaultCommander provides a default implementation of Commander.
 type DefaultCommander struct{}
 
-func (c *DefaultCommander) Install(istioCtlPath, istioOperatorPath, kubeconfigPath string) error {
-	cmd := exec.Command(istioCtlPath, "apply", "-f", istioOperatorPath, "--kubeconfig", kubeconfigPath, "--skip-confirmation")
+func (c *DefaultCommander) Install(istioOperator, kubeconfig string, logger *zap.SugaredLogger) error {
```

Comment:
The code of Install and Upgrade is pretty much the same, except the argument "upgrade" passed in this case. Maybe you can think about harmonizing this code into one new common method